pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -495,65 +499,172 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
-				if interfaceType == "virtio" {
-					interfaceType = "e1000"
-				}
-
-				// Proxy is not added as interface, Need to be added as qemu commandlist
-				// Create network interface
-				if domain.Spec.QEMUCmd == nil {
-					domain.Spec.QEMUCmd = &Commandline{}
-				}
-
-				if domain.Spec.QEMUCmd.QEMUArg == nil {
-					domain.Spec.QEMUCmd.QEMUArg = make([]Arg, 0)
-				}
-				domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: "-device"})
-				domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: fmt.Sprintf("%s,netdev=%s", interfaceType, iface.Name)})
-				// The network itself will be created on preStartHook
+			domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, domainIface)
+
+			// Create network interface
+			if domain.Spec.QEMUCmd == nil {
+				domain.Spec.QEMUCmd = &Commandline{}
 			}
 
+			if domain.Spec.QEMUCmd.QEMUArg == nil {
+				domain.Spec.QEMUCmd.QEMUArg = make([]Arg, 0)
+			}
+
+			err := createSlirpNetwork(iface, *net, domain)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+func createSlirpNetwork(iface v1.Interface, network v1.Network, domain *Domain) error {
+	qemuArg := Arg{Value: fmt.Sprintf("user,id=%s", iface.Name)}
+
+	err := configVMCIDR(&qemuArg, iface, network)
+	if err != nil {
+		return err
+	}
+
+	err = configDNSSearchName(&qemuArg)
+	if err != nil {
+		return err
+	}
+
+	err = configPortForward(&qemuArg, iface)
+	if err != nil {
+		return err
+	}
+
+	domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, Arg{Value: "-netdev"})
+	domain.Spec.QEMUCmd.QEMUArg = append(domain.Spec.QEMUCmd.QEMUArg, qemuArg)
+
+	return nil
+}
+
+func configPortForward(qemuArg *Arg, iface v1.Interface) error {
+	if iface.Slirp.Ports == nil {
+		return nil
+	}
+
+	for _, forwardPort := range iface.Slirp.Ports {
+
+		if forwardPort.Port == 0 {
+			return fmt.Errorf("Port must be configured")
+		}
+
+		// Check protocol, its case sensitive like kubernetes
+		if forwardPort.Protocol == "" {
+			forwardPort.Protocol = DefaultProtocol
+		}
+
+		// Check if PodPort is configure If not Get the same Port as the vm port
+		if forwardPort.PodPort == 0 {
+			forwardPort.PodPort = forwardPort.Port
+		}
+
+		qemuArg.Value += fmt.Sprintf(",hostfwd=%s::%d-:%d", strings.ToLower(forwardPort.Protocol), forwardPort.PodPort, forwardPort.Port)
+
+	}
+
+	return nil
+}
+
+func configVMCIDR(qemuArg *Arg, iface v1.Interface, network v1.Network) error {
+	vmNetworkCIDR := ""
+	if network.Pod.VMNetworkCIDR != "" {
+		_, _, err := net.ParseCIDR(network.Pod.VMNetworkCIDR)
+		if err != nil {
+			return fmt.Errorf("Failed parsing CIDR %s", network.Pod.VMNetworkCIDR)
+		}
+		vmNetworkCIDR = network.Pod.VMNetworkCIDR
+	} else {
+		vmNetworkCIDR = DefaultVMCIDR
+	}
+
+	// Insert configuration to qemu commandline
+	qemuArg.Value += fmt.Sprintf(",net=%s", vmNetworkCIDR)
+
+	return nil
+}
+
+func configDNSSearchName(qemuArg *Arg) error {
```

Comment:
I think you should reuse ParseSearchDomains instead of reimplementing it here.

Reply:
I cant use the function because I get a circular import.
I removed the function from the common file and pass it to the convert file.